### PR TITLE
Oas 4764 | Add Backup resource to Oasis Terraform provider

### PR DESCRIPTION
**Pull request by adoi**

### CODE REVIEW

File: examples/backup/main.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
+terraform {
+  required_version = ">= 0.13.0"
+  required_providers {
+    oasis = {
+      source = "arangodb.com/managed/oasis"
+      version = "1.5.1"
```
Comment (ewoutp) [inline examples/backup/main.tf:6]: What if we want to release a new version?
Can we make this ">= 1.5.1" ?

File: examples/backup/main.tf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  name = "oasis_test_dep_tf"
+  location {
+    region = "gcp-europe-west4"
+  }
+  version {
+    db_version = "3.8.6"
+  }
+  security {
+    disable_foxx_authentication = false
+  }
```
Comment (ewoutp) [inline examples/backup/main.tf:36]: IMO we should be able to leave this out

File: examples/deployment/main.tf [hunk 3]
```diff
@@ -30,7 +30,7 @@ resource "oasis_deployment" "my_oneshard_deployment" {
   }
   configuration {
     model = "oneshard"
-    node_size_id = "a4"
+    node_size_id = "c4-a8"
```
Comment (ewoutp) [inline examples/deployment/main.tf:33]: Why not c4-a4?

File: pkg/data_source_backup_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		backupDataSourceDescriptionFieldName:  "test-description",
+		backupDataSourceURLFieldName:          "https://test.url",
+		backupDataSourceCreatedAtFieldName:    "2022-01-01T01:01:01Z",
+		backupDataSourceDeploymentIDFieldName: "test-dep-id",
+		backupDataSourcePolicyIDFieldName:     "test-policy-id",
+	}
+
+	got := flattenBackupObject(&backup)
+	assert.Equal(t, expected, got)
+}
```
Comment (ewoutp) [inline pkg/data_source_backup_test.go:58]: Should we also have tests for the negative cases. (e.g. wrongly typed values)

Comment (adoi) [reply pkg/data_source_backup_test.go:58]: I thought about this but in the case of flattening/expanding resources we don't have validation at that level as validation is provided at schema level, or as a diff from Terraform. I have, however, added expected failure tests in resource backup when no project id is provided or there is an invalid value in the schema. For expected failure Terraform suggests to use the plan's diff, so I have added two cases when the plan is invalid. Do you think that makes sense?

File: pkg/resource_backup_test.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+    ]
+  }
+}
+
+resource "oasis_backup" "%s" {
+  name = "%s"
+  description = "test backup description update from terraform"
+  deployment_id = oasis_deployment.my_oneshard_deployment.id
+  upload = true
+  auto_deleted_at = 3
```
Comment (ewoutp) [inline pkg/resource_backup_test.go:119]: In sample it was minus 3

**Review by ewoutp — APPROVED** [review]
LGTM
